pkg/cmd/get.go

```diff
@@ -142,9 +235,144 @@ func (o *GetOptions) Complete(args []string) error {
 
+	// Pre-compute KubeArchive query parameters
+	o.kubearchiveQueryParams = o.buildKubeArchiveQueryParams()
+
 	return nil
 }
 
+// processK8sResourcesOptimized processes Kubernetes resources with optimized sorting and filtering
+func (o *GetOptions) processK8sResourcesOptimized(resources []*unstructured.Unstructured) ([]*ResourceWithAvailability, bool) {
+	result := make([]*ResourceWithAvailability, 0, min(len(resources), o.Limit))
+	var k8sTrimmed bool
+	for _, resource := range resources {
+		if !o.passesTimestampFilter(resource) {
+			continue
+		}
+
+		resourceWithAvailability := &ResourceWithAvailability{
+			Resource:  resource,
+			InCluster: true,
+			Archived:  false,
+		}
+
+		// Insert in sorted order (newest first) using binary search
+		idx := o.findInsertionIndex(result, resource)
+
+		// Insert at the correct position
+		if idx < len(result) {
+			result = append(result, nil) // Add space
+			copy(result[idx+1:], result[idx:])
+			result[idx] = resourceWithAvailability
+		} else {
+			result = append(result, resourceWithAvailability)
+		}
+
+		// Trim to limit to save memory
+		if len(result) > o.Limit {
+			result = result[:o.Limit]
+			k8sTrimmed = true
+		}
+	}
+
+	return result, k8sTrimmed
+}
+
+// mergeKubeArchiveResourcesOptimized efficiently merges KubeArchive resources with existing sorted list
+func (o *GetOptions) mergeKubeArchiveResourcesOptimized(existing []*ResourceWithAvailability, kubearchiveResources []*unstructured.Unstructured) ([]*ResourceWithAvailability, bool, bool) {
+	// Create a map of existing resources by UID for deduplication
+	var k8sTrimmed, kubeArchiveTrimmed bool
+	existingUIDs := make(map[string]int)
+	for idx, res := range existing {
+		existingUIDs[string(res.Resource.GetUID())] = idx
+	}
+
+	result := make([]*ResourceWithAvailability, 0, len(existing)+len(kubearchiveResources))
+	result = append(result, existing...)
+
+	for processedIdx, resource := range kubearchiveResources {
+		uid := string(resource.GetUID())
+
+		var insertIdx int
+		if k8sIdx, exists := existingUIDs[uid]; exists {
+			// Resource exists in both - mark as archived too
+			existing[k8sIdx].Archived = true
+		} else {
+			// Resource only in archive
+			resourceWithAvailability := &ResourceWithAvailability{
+				Resource:  resource,
+				InCluster: false,
+				Archived:  true,
+			}
+
+			// Insert in sorted order using binary search
+			insertIdx = o.findInsertionIndex(result, resource)
+
+			// Insert at the correct position
+			if insertIdx < len(result) {
+				result = append(result, nil) // Add space
+				copy(result[insertIdx+1:], result[insertIdx:])
+				result[insertIdx] = resourceWithAvailability
+			} else {
+				result = append(result, resourceWithAvailability)
+			}
+		}
+
+		// Trim to limit to save memory and stop processing if we have enough
+		if len(result) > o.Limit {
+			result = result[:o.Limit]
+			if insertIdx < len(result) {
+				k8sTrimmed = true
+			} else {
+				kubeArchiveTrimmed = true
+			}
+		}
```

Comment:
This code pretty much duplicates what is done in `processK8sResourcesOptimized`.  Can they be combined into a method?

Reply:
you are right, I've combined them in my last refactor